File: python/sorting/recursive_bubble_sort.py
''' Python Program for implementation of 
	Recursive Bubble sort ''' 

def bubble_sort(listt):
	'''Function that takes a list and Bubble sorts it recusrively'''

	for i, num in enumerate(listt): 
		try: 
			if listt[i+1] < num: 
				listt[i] = listt[i+1] 
				listt[i+1] = num 
				bubble_sort(listt) 
		except IndexError: 
			pass
	return listt 


arr = []
print("Enter Elements:")
arr = [int(num) for num in input().split()]
n = n = len(arr)
bubble_sort(arr)
print("Sorted array:")
for i in range(n):
	print(f"{arr[i]}", end = ' ')

# Code contributed by Utkarsh Bajaj

File: cpp/sorting/recursive_bubble_sort.cpp
//RECURSIVE BUBBLE SORT

#include<iostream>
using namespace std;

void recurssiveBubbleSort(int a[], int n)
{
	if (n == 1)  //base case
		return;
	
	else{
		for (int i=0; i<n-1;i++)
		{
			if(a[i]>a[i+1])
			{                     //swapping
				int temp=a[i];
				a[i]=a[i+1];
				a[i+1]=temp;
			}
		}
		
		return recurssiveBubbleSort(a,n-1);     //recurssive call
	}
}

void display(int a[], int n)
{
	cout << "\nRecursive Bubble Sort\n";
	for(int i=0;i<n;i++)
		cout << a[i] << "\t";	
}

int main()
{
	int a[50],n;
	
	cout << "Enter the length of the array ";
	cin >> n;
	
	cout << "\nEnter the elements in the array\n";
	
	for(int i=0;i<n;i++)
	{
		cin >> a[i];
	}
	
	recurssiveBubbleSort(a,n);
	display(a,n);
	return 0;
}
